lib/collections.js

```diff
@@ -157,13 +157,6 @@ var factory = module.exports = function (request) {
 
             if (query && typeof query === 'object') {
```

Comment:
You could also just move that initial definition of requestObject.endpoint down into an `} else {` clause around line 167, so it isn't executed for the case where `query` is an object.